File: 1435-xor-queries-of-a-subarray/1435-xor-queries-of-a-subarray.cpp
class Solution {
public:
    vector<int> xorQueries(vector<int>& arr, vector<vector<int>>& queries) {
        vector<int> ans, preList;
        preList.push_back(0);
        for(int i=0; i<arr.size(); i++){
            preList.push_back(preList[i]^arr[i]);
        }
        for(int i=0; i<queries.size(); i++){
            ans.push_back(preList[queries[i][1]+1]^preList[queries[i][0]]);
        }
        return ans;
    }
};

File: 1435-xor-queries-of-a-subarray/1435-xor-queries-of-a-subarray.py
class Solution:
    def xorQueries(self, arr: List[int], queries: List[List[int]]) -> List[int]:
        pre_lst=[0]
        for i in range(0,len(arr)):
            pre_lst.append(pre_lst[-1]^arr[i])
        ans=[]
        for i in queries:
            ans.append(pre_lst[i[1]+1]^pre_lst[i[0]])
        return ans